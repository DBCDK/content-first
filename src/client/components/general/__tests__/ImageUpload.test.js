/* eslint-disable no-undefined */
import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import ImageUpload from '../ImageUpload.component';
const image =
  'data:image/gif;base64,R0lGODlhAQABAIAAAP///////yH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==';
jest.mock('../../general/Spinner.component', () => 'Spinner');

describe('Image Upload', () => {
  describe('render function', () => {
    it('renders an image upload form', () => {
      const tree = renderer.create(<ImageUpload />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Add spinner on loading', () => {
      const tree = renderer.create(<ImageUpload loading={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Add preview image', () => {
      const tree = renderer
        .create(<ImageUpload previewImage={image} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Add image', () => {
      const onFile = jest.fn();
      const tree = mount(<ImageUpload previewImage={image} onFile={onFile} />);
      tree
        .find('input')
        .first()
        .simulate('change', {target: {files: [image]}});
      tree.update();
      expect(tree).toMatchSnapshot();
      expect(onFile.mock.calls[0][0]).toEqual(image);
    });
  });
});
