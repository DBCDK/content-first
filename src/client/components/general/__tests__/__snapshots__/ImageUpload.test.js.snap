// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Upload render function Add image 1`] = `
<ImageUpload
  onFile={
    [MockFunction] {
      "calls": Array [
        Array [
          "data:image/gif;base64,R0lGODlhAQABAIAAAP///////yH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  previewImage="data:image/gif;base64,R0lGODlhAQABAIAAAP///////yH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
>
  <div
    className="image-upload undefined"
  >
    <div
      className="image-upload--zone position-relative"
      style={
        Object {
          "height": "150px",
          "overflow": "hidden",
          "width": "150px",
        }
      }
    >
      <div>
        <img
          alt="User profile"
          className="preview-img w-100"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAP///////yH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
          style={
            Object {
              "opacity": 1,
              "pointerEvents": "none",
            }
          }
        />
      </div>
      <div
        className="droppable-image-overlay w-100 h-100"
      />
    </div>
    <div
      style={
        Object {
          "clear": "both",
        }
      }
    />
    <UploadButton
      className="mt-2"
      readFiles={[Function]}
    >
      <label
        className="mt-2"
      >
        <Button
          Tag="span"
          size="small"
          type="tertiary"
        >
          <span
            className="
      button button__small button__tertiary   "
            data-cy=""
            disabled={false}
          >
            <span>
              <span
                className="align-middle"
              >
                Upload billede
              </span>
            </span>
          </span>
        </Button>
        <input
          accept="image/png, image/jpeg"
          className="droppable-image-field--file-input"
          onChange={[Function]}
          type="file"
        />
      </label>
    </UploadButton>
    <Error />
  </div>
</ImageUpload>
`;

exports[`Image Upload render function Add preview image 1`] = `
<div
  className="image-upload undefined"
>
  <div
    className="image-upload--zone position-relative"
    style={
      Object {
        "height": "150px",
        "overflow": "hidden",
        "width": "150px",
      }
    }
  >
    <div>
      <img
        alt="User profile"
        className="preview-img w-100"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAP///////yH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
        style={
          Object {
            "opacity": 1,
            "pointerEvents": "none",
          }
        }
      />
    </div>
    <div
      className="droppable-image-overlay w-100 h-100"
    >
      
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  />
  <label
    className="mt-2"
  >
    <span
      className="
      button button__small button__tertiary   "
      data-cy=""
      disabled={false}
    >
      <span>
        <span
          className="align-middle"
        >
          Upload billede
        </span>
      </span>
    </span>
    <input
      accept="image/png, image/jpeg"
      className="droppable-image-field--file-input"
      onChange={[Function]}
      type="file"
    />
  </label>
</div>
`;

exports[`Image Upload render function Add spinner on loading 1`] = `
<div
  className="image-upload undefined"
>
  <div
    className="image-upload--zone position-relative"
    style={
      Object {
        "height": "150px",
        "overflow": "hidden",
        "width": "150px",
      }
    }
  >
    <Spinner
      size="50px"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "margin": "auto",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <div
      className="d-flex align-items-center justify-content-center h-100 "
    >
      <i
        aria-label=""
        className="material-icons material-icons-image md-xxlarge "
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#xE3F4;",
          }
        }
        data-unicoded="true"
        onClick={[Function]}
        role=""
      />
    </div>
    <div
      className="droppable-image-overlay w-100 h-100"
    >
      
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  />
  <label
    className="mt-2"
  >
    <span
      className="
      button button__small button__tertiary   "
      data-cy=""
      disabled={false}
    >
      <span>
        <span
          className="align-middle"
        >
          Upload billede
        </span>
      </span>
    </span>
    <input
      accept="image/png, image/jpeg"
      className="droppable-image-field--file-input"
      onChange={[Function]}
      type="file"
    />
  </label>
</div>
`;

exports[`Image Upload render function renders an image upload form 1`] = `
<div
  className="image-upload undefined"
>
  <div
    className="image-upload--zone position-relative"
    style={
      Object {
        "height": "150px",
        "overflow": "hidden",
        "width": "150px",
      }
    }
  >
    <div
      className="d-flex align-items-center justify-content-center h-100 "
    >
      <i
        aria-label=""
        className="material-icons material-icons-image md-xxlarge "
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#xE3F4;",
          }
        }
        data-unicoded="true"
        onClick={[Function]}
        role=""
      />
    </div>
    <div
      className="droppable-image-overlay w-100 h-100"
    >
      
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  />
  <label
    className="mt-2"
  >
    <span
      className="
      button button__small button__tertiary   "
      data-cy=""
      disabled={false}
    >
      <span>
        <span
          className="align-middle"
        >
          Upload billede
        </span>
      </span>
    </span>
    <input
      accept="image/png, image/jpeg"
      className="droppable-image-field--file-input"
      onChange={[Function]}
      type="file"
    />
  </label>
</div>
`;
